agent/local.go

```diff
@@ -91,12 +92,6 @@ func (l *localState) Init(config *Config, logger *log.Logger) {
 	l.triggerCh = make(chan struct{}, 1)
 }
 
-// SetIface is used to set the Consul interface. Must be set prior to
-// starting anti-entropy
-func (l *localState) SetIface(iface consul.Interface) {
```

Comment:
(no change requested, commenting for my understanding) - this looks like it was a brittle way to initialize the delegate, that required this method to be called separate from the already existing Init(..) method, now you've made them all to be initialized together when Init(..) is called, which is cool